### PR TITLE
added `PIDLoop` interface

### PR DESCRIPTION
**Pull request by Ergoold**

### CODE REVIEW

**Comment by Ergoold** [comment]
[Add A PIDLoop Interface](https://trello.com/c/JU0zMz21/11-add-a-pidloop-interface)

**Comment by Ergoold** [comment]
[Add A PIDLoop Interface](https://trello.com/c/JU0zMz21/11-add-a-pidloop-interface)

**Comment by Ergoold** [comment]
[Add A PIDLoop Interface](https://trello.com/c/JU0zMz21/11-add-a-pidloop-interface)

File: src/com/spikes2212/control/PIDLoop.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.spikes2212.control;
+
+/**
+ * An interface for PID loop controllers.
```
Comment (ScarletMetal) [inline src/com/spikes2212/control/PIDLoop.java]: replace `controllers` with `loops`

File: src/com/spikes2212/control/PIDLoop.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ *
+ * @author Eran Goldstein
+ */
+public interface PIDLoop {
+    /**
+     * Starts running the PID loop.
```
Comment (ScarletMetal) [inline src/com/spikes2212/control/PIDLoop.java]: remove the word `running`

File: src/com/spikes2212/control/PIDLoop.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+     */
+    void enable();
+
+    /**
+     * Stops running the PID loop.
```
Comment (ScarletMetal) [inline src/com/spikes2212/control/PIDLoop.java]: remove the word `running`

File: src/com/spikes2212/control/PIDLoop.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.spikes2212.control;
+
+/**
+ * An interface for PID loop controllers.
+ *
+ * @author Eran Goldstein
+ */
+public interface PIDLoop {
```
Comment (ScarletMetal) [inline src/com/spikes2212/control/PIDLoop.java]: please add a `void updateSetpoint()` and `boolean onTarge()` methods that would behave like the ones in the old `PIDController`

**Pull request closed without merging**